### PR TITLE
New Whitelist / Blacklist Filter

### PR DESCRIPTION
**Pull request by gsomoza**
This new filter will behave either as a blacklist or a whitelist. If the value being filtered exists in the list of known values, the filter will return the value if its a whitelist or `null` if its a blacklist. The opposite will happen if value is not in the list.
### Whitelist

``` php
$whitelist = new \Zend\Filter\Whitelist(array(
    'list' => array('allowed-1', 'allowed-2')
));
echo $whitelist->filter('allowed-2');   // => 'allowed-2'
echo $whitelist->filter('not-allowed'); // => null
```
### Blacklist

``` php
$blacklist = new \Zend\Filter\Whitelist(array(
    'type' => 'blacklist', // optional, default is 'whitelist'
    'list' => array('not-allowed-1', 'not-allowed-2')
));
echo $blacklist->filter('allowed');        // => 'allowed'
echo $blacklist->filter('not-allowed-2');  // => null
```
### Notes
1. Maybe calling the filter class `Whitelist` might be counter-intuitive if the desired behavior is that of a blacklist. But I couldn't think of a better way to name the filter, so I'm open to suggestions:
   - Something like "AllowedList" would just be a synonym of "whitelist". 
   - "List" would just be too vague.
   - Separate filter classes seemed overkill.
2. I decided to allow only one type of configuration in the constructor (by array of options) just to keep it simple. If you'd like to allow for a more dynamic constructor (like a few other filters do) let me know.
3. Not sure what commit edbc12b is doing in this PR. Looks like its in `master` but not in `develop`.

See #6960 for more info.

### CODE REVIEW

File: library/Zend/Filter/Blacklist.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline library/Zend/Filter/Blacklist.php]: 2015 ;)

File: library/Zend/Filter/Whitelist.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline library/Zend/Filter/Whitelist.php]: 2015 ;)

File: tests/ZendTest/Filter/BlacklistTest.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline tests/ZendTest/Filter/BlacklistTest.php]: 2015 ;)

File: tests/ZendTest/Filter/WhitelistTest.php [hunk 4]
```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline tests/ZendTest/Filter/WhitelistTest.php]: 2015 ;)

**Comment by weierophinney** [comment]
Merged -- now I need @gabrielsomoza to write up docs on the new filter and send a PR to https://github.com/zendframework/zf2-documentation :)

**Comment by gabrielsomoza** [comment]
Sure, will do - any chance you could give me a deadline? (its easier for me if I can schedule it :) )

**Comment by Ocramius** [comment]
@gabrielsomoza should be ~2 weeks

**Pull request merged**